app/services/banking/bank_transactions_trimmer.rb

```diff
@@ -0,0 +1,18 @@
+module Banking
+  class BankTransactionsTrimmer
+    def self.call(legal_aid_application)
+      new(legal_aid_application).call
+    end
+
+    def initialize(legal_aid_application)
+      @legal_aid_application = legal_aid_application
+    end
+
+    def call
+      deletion_period = (@legal_aid_application.transaction_period_finish_on..1.year.from_now)
```

Comment:
Just to confirm, transaction_period_finish_on can be the used delegated functions date or todays date?

And that here, you destroy anything past it.

Since the used delegated functions date can't be more than 12 months ago, the  `..1.year.from_now`  ensures that all transactions are deleted after that.

Reply:
Should it be `1.year.from_now` ... or `Date.current` (today) ?

Reply:
We destroy anything past the end of the calculation period, which is either the day delegated functions were used, or midnight last night.  That is because we now collect ALL transactions from the beginning of the calculation period to NOW, in order that we can get the latest balance and that ties in with the running balance on the last transaction if present, and we can calculate it if not.

But after we've done that, we don't need the bank transactions after the end of the calculation period to go into the CSV file, which is why we delete them.